sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/AddressSelector.java

```diff
@@ -45,6 +46,20 @@ public Mono<Uri> resolvePrimaryUriAsync(RxDocumentServiceRequest request, boolea
         });
     }
 
+    public Mono<Uri> resolvePrimaryUriAsync(RxDocumentServiceRequest request, boolean forceAddressRefresh, Map<Uri, String> replicaStatuses) {
```

Comment:
the other resolvePrimaryUriAsync method is only used in tests method? If that is the case, we can remove it

Reply:
It is not only used in tests. It is used when performing a background refresh of the primary address.